server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/service/ExpireMailboxService.java

```diff
@@ -187,7 +188,8 @@ public Mono<Result> expireMailboxes(Context context, RunningOptions runningOptio
                         SearchQuery.headerDateBefore("Expires", now, DateResolution.Second)
                     )
             );
-            return Iterators.toFlux(usersRepository.list())
+            // Note: user list may be a blocking iterable, must run on a scheduler that supports this.
+            return Iterators.toFlux(usersRepository.list()).subscribeOn(Schedulers.elastic())
```

Comment:
```suggestion
            return Iterators.toFlux(usersRepository.list()).subscribeOn(ReactorUtils.BLOCKING_CALL_WRAPPER)
```

 -> elastic is deprecated in favor of bounded elastic
 -> nested calls to bounded elastic causes dead locks hence when we know we would end up calling .block or equivalent we need to use a distinct scheduler.
 
Having a reactive version of userRepository.list would be handy!

Reply:
Thanks for the feedback, I am still tripping over the fine details of reactor, sigh. Fixed in latest push.